hibernate-redis/src/main/java/org/hibernate/cache/redis/util/JedisTool.java

```diff
@@ -104,39 +104,38 @@ private static Properties loadCacheProperties(final Properties props) {
                 try { is.close(); } catch (Exception ignored) { }
             }
         }
+        // now add and override with any settings from the cache_provider_config
+        cacheProps.putAll(fileSystemProperties);
         return cacheProps;
     }
 
     /**
      * Get expire time out for the specified region
      *
-     * @param regionName    region name defined at Entity
-     * @param defaultExpiry default expiry in seconds
+     * @param props      properties containing expiration settings
+     * @param regionName region name defined at Entity
      * @return expiry in seconds
      */
-    public static int getExpireInSeconds(final String regionName, final int defaultExpiry) {
-        if (cacheProperties == null)
+    public static int getExpireInSeconds(final Properties props, final String regionName) {
+        int defaultExpiry = getDefaultExpireInSeconds(props);
+        if (props == null)
             return defaultExpiry;
-        return Integer.valueOf(getProperty(EXPIRY_PROPERTY_PREFIX + regionName, String.valueOf(defaultExpiry)));
+        int expireInSeconds = Integer.valueOf(props.getProperty(EXPIRY_PROPERTY_PREFIX + regionName, String.valueOf(defaultExpiry)));
+        log.debug("getExpireInSeconds. regionName=[{}], expireInSeconds=[{}]",
+                regionName, expireInSeconds);
+        return expireInSeconds;
     }
 
     /**
-     * retrieve property value in hibernate-redis.properties
+     * Get the default expire time from the supplied properties
      *
-     * @param name         property key
-     * @param defaultValue default value
-     * @return property value
+     * @param props   properties containing expiration settings
+     * @return expiry in seconds
      */
-    public static String getProperty(final String name, final String defaultValue) {
-        if (cacheProperties == null)
-            return defaultValue;
-        try {
-            String value = cacheProperties.getProperty(name, defaultValue);
-            log.debug("get property. name=[{}], value=[{}], defaultValue=[{}]", name, value, defaultValue);
-            return value;
-        } catch (Exception ignored) {
-            return defaultValue;
-        }
-    }
+    private static int getDefaultExpireInSeconds(final Properties props) {
+        if (props == null)
+            return JedisClient.DEFAULT_EXPIRY_IN_SECONDS;
 
+        return Integer.decode(props.getProperty(EXPIRE_IN_SECONDS, String.valueOf(JedisClient.DEFAULT_EXPIRY_IN_SECONDS)));
```

Comment:
Its not a regression in the sense that the old code was never going to catch a NumberFormatException. I'm not sure what the try catch above was ever really doing. I don't think getProperty can throw.

I agree we could/should try to catch the NumberFormatException but I'm not hugely worried about it. It will fail spectacularly and i'm not sure you want to swallow the exception and just rely on the default if you miss configure things.